app/code/Magento/CatalogStorefront/Test/Api/ProductVariants/ConfigurableVariantsTest.php

```diff
@@ -109,19 +117,75 @@ public function testConfigurableProductVariants(): void
      */
     public function testProductVariantsDisabledProduct(): void
     {
+        //This sleep ensures that the elastic index has sufficient time to refresh
+        //See https://www.elastic.co/guide/en/elasticsearch/reference/6.8/docs-refresh.html#docs-refresh
+        sleep(1);
         /** @var $configurable Product */
         $configurable = $this->productRepository->get('configurable');
 
         $this->variantsRequestInterface->setProductId((string)$configurable->getId());
         $this->variantsRequestInterface->setStore('default');
+        /** @var $variantServiceItem ProductVariantResponse */
+        $variantServiceItem = $this->variantService->getProductVariants($this->variantsRequestInterface);
+        $actual = $this->responseArrayMapper->convertToArray($variantServiceItem)['matched_variants'];
+        self::assertCount(1, $actual);
+    }
 
+    /**
+     * Validate matching of variants by option values
+     *
+     * @magentoApiDataFixture Magento/ConfigurableProduct/_files/configurable_product_nine_simples.php
+     * @magentoDbIsolation disabled
+     * @throws NoSuchEntityException
+     * @throws \Throwable
+     */
+    public function testVariantsMatch(): void
+    {
         //This sleep ensures that the elastic index has sufficient time to refresh
         //See https://www.elastic.co/guide/en/elasticsearch/reference/6.8/docs-refresh.html#docs-refresh
         sleep(1);
+        $simpleSkus = [
+            'simple_0',
+            'simple_1',
+            'simple_2',
+            'simple_3',
+            'simple_4',
+            'simple_5',
+            'simple_6',
+            'simple_7',
+            'simple_8'
+        ];
+        /** @var $configurable Product */
+        $configurable = $this->productRepository->get('configurable');
+        $simples = [];
+        foreach ($simpleSkus as $sku) {
+            $simples[] = $this->productRepository->get($sku);
+        }
+        $availableVariants = $this->getExpectedProductVariants($configurable, $simples);
+        self::assertCount(9, $availableVariants, 'Wrong number of expected variants');
```

Comment:
We don't test here our API, this part already been tested by Magento. If this error occurs "Wrong number of expected variants" we would guess we have an issue with API, but the problem will be with Magento repositories.

Reply:
Removed this part. We will get null pointer anyway if its empty, but I guess that's an error and not test failure.